server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -388,10 +403,21 @@ public void writeText(Text text) throws IOException {
     }
 
     public void writeString(String str) throws IOException {
+        writeString(str, scratch.get(), 0);
+    }
+
+    /**
+     * Write string as well as possibly the beginning of the given {@code buffer}. The given {@code buffer} will also be used when encoding
+     * the given string.
+     *
+     * @param str string to write
+     * @param buffer buffer that may hold some bytes to write
+     * @param off how many bytes in {code buffer} to write
```

Comment:
This is confusing to me; isn't this the `offset` at which writing happens, not how many bytes to write? This string is confusing.